sidecar/config/config.go

```diff
@@ -139,9 +125,14 @@ type Config struct {
 	Service  Service  `yaml:"service"`
 	Endpoint Endpoint `yaml:"endpoint"`
 
-	Registry   Registry   `yaml:"registry"`
-	Controller Controller `yaml:"controller"`
-	Dnsconfig  Dnsconfig  `yaml:"dnsconfig"`
+	DiscoveryType string `yaml:"discovery_type"`
+	RulesType     string `yaml:"rules_type"`
```

Comment:
Despite our discussion yesterday, I think that type is not the best term to use here.
`rules_type: amalgam8` or `rules_type: kubernetes` sounds a bit weird. `discovery_type` somehow makes more sense.
All in all, I think `rules_backend: kubernetes` and `discovery_backend: kubernetes` would be a better fit.